### ISSUE TITLE
self_check_output / output flows don't activate

### ISSUE THREAD

**Issue by dannielvieira-hotmart** [NVIDIA/NeMo-Guardrails#270]
I'm running the getting_started notebooks to experiment and understand how each Rails work, but none of the output rails declared work. For example, on the [https://github.com/NVIDIA/NeMo-Guardrails/tree/develop/docs/getting_started/5_output_rails](Output Rail) introuction notebook, when running the section that uses the output flow declared in the config.yml, it doesn't get activated when calling the rails.generate. Why is that?

Prints of the llm_calls_summary result, config.yml and prompts.yml. 
I followed along the tutorial and got to this, does anyone what happened?

<img width="1076" alt="image" src="https://github.com/NVIDIA/NeMo-Guardrails/assets/120054081/1263fd47-0ac4-466e-857c-ed6d4e56d762">
<img width="1223" alt="image" src="https://github.com/NVIDIA/NeMo-Guardrails/assets/120054081/cbf1e532-9160-4b7a-a3b5-1e0a6affe99b">
<img width="1223" alt="image" src="https://github.com/NVIDIA/NeMo-Guardrails/assets/120054081/6af4bc68-70bf-42ef-b920-c0a945efe78e">

**Comment by drazvan** [comment]
Hi @dannielvieira-hotmart! It seems like it's somehow still using the config from the previous step, where the output rail was not configured. Can you make sure all the steps from the notebook run in order. Also, for a sanity check, do also add the following before the print statement: 
```
print(rails.config.rails)
print(info.llm_calls[2].prompt)
```

You should see something similar to
```
config=RailsConfigData(fact_checking=FactCheckingRailConfig(parameters={}, fallback_to_self_check=False), sensitive_data_detection=SensitiveDataDetection(recognizers=[], input=SensitiveDataDetectionOptions(entities=[], mask_token='*'), output=SensitiveDataDetectionOptions(entities=[], mask_token='*'), retrieval=SensitiveDataDetectionOptions(entities=[], mask_token='*'))) input=InputRails(flows=['self check input']) output=OutputRails(flows=['self check output', 'check blocked terms']) retrieval=RetrievalRails(flows=[]) dialog=DialogRails(single_call=SingleCallConfig(enabled=False, fallback_to_multiple_calls=True), user_messages=UserMessagesConfig(embeddings_only=False))
```

**Comment by dannielvieira-hotmart** [comment]
Hello @drazvan. I restarted the kernel and ran in order, but still somehow is not loading current folder config, in it. Here is the output of the `rails.config.rails`:

<img width="1220" alt="image" src="https://github.com/NVIDIA/NeMo-Guardrails/assets/120054081/f8e5c2d9-8394-4f49-9811-4b9b3c09cc3b">

I tested same notebook, but running on a different kernel and folder, and it ran just fine. Really can't tell why is using previous step.
